src/prototype_room_init.js

```diff
@@ -97,11 +133,6 @@ function setStructures(room, path, costMatrixBase) {
         costMatrixBase.set(structurePos.x, structurePos.y, config.layout.structureAvoid);
         continue;
       }
-      if (room.memory.position.structure.lab.length < CONTROLLER_STRUCTURES.lab[8]) {
-        room.memory.position.structure.lab.push(structurePos);
-        costMatrixBase.set(structurePos.x, structurePos.y, config.layout.structureAvoid);
-        continue;
-      }
```

Comment:
I'm tending to keep this. You set the structures before, so nothing should happen here.

If the mineral path is too short or whatever, it would place the rest of the labs and terminal here.